engine/common/multiplexer/engine.go

```diff
@@ -0,0 +1,175 @@
+package multiplexer
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network"
+	"github.com/rs/zerolog"
+)
+
+type Engine struct {
```

Comment:
To make the separation of concerns more clear, I would recommend we have a `MultiplexerNetwork` and a `MultiplexerEngine` (instead of a single Engine which has both).
The `MultiplexerNetwork` contains a reference to the `MultiplexerEngine`, the default staked network, the new unstaked network and implements the `Register` method. 
The `MultiplexerNetwork` can then be passed to the other engines which need a network e.g. the IngestionEngine (instead of passing the standard Network).
This Engine seems to mix the functionality of a Network and an Engine and if kept the separate it will be easy to test them.

Reply:
Good idea, will change that.

Reply:
@vishalchangrani Updated.